quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/infinispan/CacheManagerFactory.java

```diff
@@ -355,11 +375,34 @@ private static int getStartTimeout() {
         return Integer.getInteger("kc.cache-ispn-start-timeout", 120);
     }
 
-    private static void configureTransportStack(ConfigurationBuilderHolder builder) {
+    private static void configureTransportStack(ConfigurationBuilderHolder builder, KeycloakSession keycloakSession) {
         String transportStack = Configuration.getRawValue("kc.cache-stack");
 
         var transportConfig = builder.getGlobalConfigurationBuilder().transport();
         if (transportStack != null && !transportStack.isBlank()) {
+            if ("jdbc-ping".equals(transportStack)) {
+                EntityManager em = keycloakSession.getProvider(JpaConnectionProvider.class).getEntityManager();
```

Comment:
@ryanemerson Would this line be enough to make sure the tables are created at the time jdbc ping is initialized? Maybe we don't need anything to change in `quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/QuarkusKeycloakSessionFactory.java`

Reply:
Unfortunately not. Removing my hack means that the `EntityManagerFactory` in the can be null and we get the following fatal exception:

```
2024-10-09 11:55:55,629 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (development) mode
2024-10-09 11:55:55,630 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Cannot invoke "jakarta.persistence.EntityManagerFactory.createEntityManager(jakarta.persistence.SynchronizationType)" because "emf" is null
```